### ISSUE TITLE
can not map \\wsl$\xxx\xxx path to a network driver 

### ISSUE THREAD

**Issue by ijesonchen** [microsoft/WSL#4657]
Please use the following bug reporting template to help produce issues which are actionable and reproducible, including **all** command-line steps necessary to induce the failure condition. Please fill out **all** the fields! Issues with missing or incomplete issue templates will be closed.

If this is a console issue (a problem with layout, rendering, colors, etc.), please post to the [console issue tracker](https://github.com/microsoft/console/issues).

**Important: Do not open GitHub issues for Windows crashes (BSODs) or security issues.**  Please direct all Windows crashes and security issues to secure@microsoft.com.  Ideally, please [configure your machine to capture minidumps](https://support.microsoft.com/en-us/help/315263/how-to-read-the-small-memory-dump-file-that-is-created-by-windows-if-a), repro the issue, and send the minidump from "C:\Windows\minidump\".\\

See [our contributing instructions](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md) for assistance.

**Please fill out the below information:**
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
10.0.19013.1
wsl2 with ubuntu 18.04

* What you're doing and what's happening:  (Copy&paste the full set of _specific_ command-line steps necessary to reproduce the behavior, and their output. Include screen shots if that helps demonstrate the problem.)

1. in explorer, mapping a network driver with path \\wsl$\Ubuntu-18.04 succeeded,
2. in console, "dir \\wsl$\Ubuntu-18.04\home" can display directory correctly.
3. in explorer mapping a network driver with path \\wsl$\Ubuntu-18.04\home failed, with error windows can not access \\wsl$\Ubuntu-18.04\home

* What's wrong / what should be happening instead:
mapping \\wsl$\Ubuntu-18.04\home a network driver should succeed

* Strace of the failing command, if applicable:  (If `some_command` is failing, then run `strace -o some_command.strace -f some_command some_args`, and link the contents of `some_command.strace` in a [gist](https://gist.github.com/) here).

* For WSL launch issues, please [collect detailed logs](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs).

**Comment by therealkenc** [comment]
Ref #3854. Could you please rephrase this as a step of Powershell [CLI](https://en.wikipedia.org/wiki/Command-line_interface) steps to be clear what you are doing.

**Comment by dustyjewett** [comment]
Not the author, but I was able to reproduce this. The general idea is that you can map to the root folder of a WSL, but not any deeper.  The author is able to map `\\wsl$\Ubuntu` to a drive, but if you add the `\home`, the mapping fails.

I don't believe there's any way to map drives through powershell.

**Comment by therealkenc** [comment]
Alright.... for lack...

![image](https://user-images.githubusercontent.com/7049700/68523203-9c486300-026a-11ea-9b5a-932039ad58e1.png)

**Comment by Tulitomaatti** [comment]
I'm experiencing the same limitation on 10.0.19030.1 & wsl2 with Ubuntu 18.04.

**Comment by Vaelek** [comment]
Any progress on this? A year later and it still persists.

**Comment by lkyu-ly** [comment]
> Alright.... for lack...  好吧……因为缺乏……

On latest win11, I can't use "Map network drive" to add "\\wsl.localhost\Debian"(my distro) as a drive. However when I searched this, those who use win10 would success.

What's more, although execute this command `net use Z: \\wsl.localhost\Debian` returns a succees, there's no Z: in my explorer.
Confusing.

![Image](https://github.com/user-attachments/assets/24a47d5d-1e36-4c56-9cfa-085ec21491b1)
